pre_commit_hooks/check_added_large_files.py

```diff
@@ -21,11 +21,19 @@ def lfs_files() -> Set[str]:
     return set(json.loads(lfs_ret)['files'])
 
 
-def find_large_added_files(filenames: Sequence[str], maxkb: int) -> int:
+def find_large_added_files(
+    filenames: Sequence[str], maxkb: int,
+    enforce_all: Optional[bool] = False,
```

Comment:
`Optional[bool]` means `Union[bool, None]` -- you don't want `Optional` here, just `bool`

I would make this a keyword-only argument to enforce that the calling construct is readable -- otherwise this reads like a [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap)

(not a big deal, but I usually double-indent parameters to differentiate them from the function body, but yeah I know I don't have a formatter -- I wouldn't have mentioned this if I wasn't commenting on the PR for other reasons)

Reply:
Thanks for the link on api design.